[SYCL] Updated atomic_memory_order_capabilities query test for OpenCL 3.0 backend

andylshort opened the pull request:
Both `context` and `device` should return at least `memory_order::relaxed` now on implementation-side in https://github.com/intel/llvm/pull/8517, so this test should pass if the target device has an OpenCL 3.0 backend or newer.

Remaining work/discussions required:
1. How should the test be structured for Level Zero and OpenCL <3.0 backends? These don't support the query presently (subject to change?) so this test can fail on older OCL backends currently.
2. `atomic_memory_order_acq_rel.cpp` and `atomic_memory_order_seq_cst.cpp` require updating once the query fully supports the rest of the possible values, also.

Commit:
Test OpenCL backend for atomic_memory_order_capabilities

Both context and device should return at least memory_order::relaxed now on
implementation-side, so this test should pass if the target device has
an OpenCL 3.0 backend or newer.

andylshort commented:
/verify with https://github.com/intel/llvm/pull/8517

Commit:
Add level_zero and opencl backends to atomic_memory_order* tests

Need to adjust figures in acq_rel test to reduce work-group size so test
can be used with all backends and devices.

steffenlarsen commented on SYCL/AtomicRef/atomic_memory_order_seq_cst.cpp:
```diff
@@ -1,10 +1,7 @@
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -O3 -o %t.out -Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_70
```
Some other tests have these despite being run on other backends too. Can we keep them without the tests failing? I should help ensure the CUDA backend supports the tests.

andylshort replied:
They didn't fail with these, so I shall re-add them

Commit:
Re-added cuda-specific test details

andylshort commented:
One final question to clarify something: The acq_rel test, as it stands (I've not modified the functionality, just added the level-zero and OpenCl backends to check) will fail if the target OCL backend implementation is <2.0 but should work otherwise. I've asked around and there doesn't _yet_ seem to be a way to test or filter out certain OpenCL versions, but don't want to leave it untested for relevant/newer implementations. 

Is it fine to merge as it stands? Would it affect the CI? I'm not sure what version of OpenCL the runners use.

Commit:
Merge branch 'intel' into alamzeds/atomic_mem_orders_caps_fix_test

Commit:
Merge branch 'intel' into alamzeds/atomic_mem_orders_caps_fix_test

Commit:
Reduced work group size for acq_rel test

Passes on cpu, fails on GPU, so I think a problem exists with the GPU driver
implementation, maybe specifically with fetch_add.

Commit:
Merge branch 'alamzeds/atomic_mem_orders_caps_fix_test' of https://github.com/andylshort/llvm-test-suite into alamzeds/atomic_mem_orders_caps_fix_test

andylshort commented:
I've reduced the work group size for the `acq_rel` test. It works now, but still fails on GPU targets. I believe there may be an issue in the driver implementation for GPUs - on both OpenCl and Level Zero. Likely with `fetch_add` or `atomic_ref` in general.
